cmsController.php

```diff
@@ -0,0 +1,195 @@
+                               =`images`.`id`
+                               WHERE `title`='$selectedItem' && `title` != 'submitPF';");
+    $query->execute();
+    return $query->fetchall();
+}
+
+/**
+ * Gets list of articles from DB
+ *
+ * @param $db PDO to select from
+ * @return array of articles
+ */
+function ArticleList (PDO $db):array {
+    $query=$db->prepare("SELECT `title` FROM `articles` WHERE `deleted`!=1 ");
+    $query->execute();
+    return $query->fetchall();
+}
+
+/**
+ * Takes selected article POSTdata and selects values from DB based on POST value
+ *
+ * @param $db PDO to select from
+ * @param $postData assoc array userresponse for article to select
+ * @return array
+ */
+function SelectArt (PDO $db, array $postData):array {
+    $artSelect = $postData['artSelect'];
+    $query = $db->prepare("SELECT `id`,`title`,`description`, `URL`
+                               FROM `articles`
+                               WHERE `title`='$artSelect';");
+    $query->execute();
+    return $query->fetchall();
+}
+
+/**
+ * Updates About sections on DB
+ *
+ * @param $postData Array used to test if function should run and informs what content should be added to DB
+ * @param $db to add to
+ */
+function updateAbout (array $postData,PDO $db) {
+    if(array_key_exists('submitAbout',$postData)) {
+        foreach($postData as $keys=>$values) {
+            $query = $db->prepare("REPLACE INTO `staticContent`(`name`,`content`) VALUES (?,?)");
+            $query->execute([$keys, $values]);
+        }
+    }
+}
+
+/**
+ * Updates/adds Portfolio item based on Name
+ *
+ * @param $postData array used to test if function should run and informs what content should be added to DB
+ * @param $db to add to
+ */
+function updatePortfolio (array $postData,PDO $db) {
+    if(array_key_exists('submitPf',$postData)) {
+        $query = $db->prepare("REPLACE INTO `portfolioItems`(`title`,`description`,`imgRef`,`projURL`,`github`)
+                                     VALUES (:title, :descr, :picRef, :url, :github);");
+        $query->bindValue(':title', $postData['pfTitle']);
+        $query->bindValue(':descr', $postData['pfDesc']);
+        $query->bindValue(':url', $postData['pfURL']);
+        $query->bindValue(':github', $postData['githubURL']);
+        $query->bindValue(':picRef', $postData['picSelect']);
+        $query->execute();
+    }
+}
+
+/**
+ * Takes list of Image names from DB and returns an array
+ *
+ * @param $db PDO to select from
+ * @return array of images, values
+ */
+function getImgDropDown (PDO $db):array {
+    $query=$db->prepare("SELECT `id`,`name` FROM `images` WHERE `deleted` !=1;");
+    $query->execute();
+    $items=$query->fetchall();
+    return $items;
+}
+
+/**
+ * Takes array of images and creates a string of HTML options for each image
+ *
+ * @param $arr Array of images, values
+ *
+ * @return string of HTML options for images
+ */
+function makeImgDropDown (array $arr):string {
```

Comment:
what is $arr? variable name means nothing

Reply:
resolved